src/cpsr.rs

```diff
@@ -1,51 +1,293 @@
+
+    /// The interrupt bit I is used to disable IRQ interrupts respectively (1 means disabled)
+    pub(crate) fn set_irq_disable(&mut self) {
+        // TODO: Should we check we are in privileged modes or it occurred an exeption?
+        self.0 |= 0b0000_0000_0000_0000_0000_0000_1000_0000;
+    }
+
+    /// The interrupt bit I is used to enable IRQ interrupts respectively (0 means enabled)
+    pub(crate) fn set_not_irq_disable(&mut self) {
+        // TODO: Should we check we are in privileged modes or it occurred an exeption?
+        self.0 &= 0b1111_1111_1111_1111_1111_1111_0111_1111;
+    }
+
+    /// The interrupt bit F is used to disable FIQ interrupts respectively (1 means disabled)
+    pub(crate) fn set_fiq_disable(&mut self) {
+        // TODO: Should we check we are in privileged modes or it occurred an exeption?
+        self.0 |= 0b0000_0000_0000_0000_0000_0000_0100_0000;
+    }
+
+    /// The interrupt bit F is used to enable FIQ interrupts respectively (1 means disabled)
+    pub(crate) fn set_not_fiq_disable(&mut self) {
+        // TODO: Should we check we are in privileged modes or it occurred an exeption?
+        self.0 &= 0b1111_1111_1111_1111_1111_1111_1011_1111;
+    }
+
+    /// The T Bit is used to set the current state of the CPU on THUMB mode
+    pub(crate) fn set_state_bit(&mut self) {
+        // TODO: Must be changeed by BX instructions
+        self.0 |= 0b0000_0000_0000_0000_0000_0000_0010_0000;
+    }
+
+    /// The T Bit is used to set the current state of the CPU on ARM mode
+    pub(crate) fn set_not_state_bit(&mut self) {
+        // TODO: Must be changeed by BX instructions
+        self.0 &= 0b1111_1111_1111_1111_1111_1111_1101_1111;
+    }
+
+    /// The Mode Bits M4-M0 contain the current operating mode
+    pub(crate) fn set_mode_bits(&mut self, control_bits: ModeBits) {
+        match control_bits {
+            ModeBits::OldUser => self.0 |= 0b0000_0000_0000_0000_0000_0000_0000_0000,
+            ModeBits::OldFiq => self.0 |= 0b0000_0000_0000_0000_0000_0000_0000_0001,
+            ModeBits::OldIrq => self.0 |= 0b0000_0000_0000_0000_0000_0000_0000_0010,
+            ModeBits::OldSupervisor => self.0 |= 0b0000_0000_0000_0000_0000_0000_0000_0011,
+            ModeBits::User => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_0000,
+            ModeBits::Fiq => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_0001,
+            ModeBits::Irq => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_0010,
+            ModeBits::Supervisor => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_0011,
+            ModeBits::Abort => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_0111,
+            ModeBits::Undefined => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_1011,
+            ModeBits::System => self.0 |= 0b0000_0000_0000_0000_0000_0000_0001_1111,
+        }
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::cpsr::Cpsr;
+    use crate::{condition::ModeBits, cpsr::Cpsr};
```

Comment:
Change with `super::*` is easier for unit tests

Reply:
:)